File: example5/src/main/java/ru/example/example5/EncoderJob.java
package ru.example.example5;

/**
 * Работа, в которой выполняется кодирование аудио-видео.
 * Результат работы передаёт Логеру
 */
public class EncoderJob extends JobProducer.IJob {
    private final JobLogger logger;

    protected EncoderJob(int no, JobLogger logger) {
        super(no);
        this.logger = logger;
    }

    @Override
    public void work() {
        System.out.println("job-" + no + " start in thread " + Thread.currentThread().getName());
        logger.postResult("job-" + no + " work ok");
        System.out.println("job end");
    }
}

File: example5coroutines/src/main/java/ru/example/example5coroutines/EncoderJob.kt
package ru.example.example5coroutines

/**
 * Выполняет работу в другом потоке, например кодирование аудио/видео
 */
class EncoderJob(no: Int, val logger: JobLogger) : JobProducer.IJob(no) {
    override suspend fun work() {
        println("job start #$no on thread ${Thread.currentThread().name}")
        // logger
        logger.postResult("job #$no OK")
        println("job end #$no")
    }
}